### DISCUSSION TITLE
New Apple requirements for user defaults

### DISCUSSION THREAD

**Discussion by russhwolf** [russhwolf/multiplatform-settings#164] [General]
Apple will soon require declared reasons for using certain APIs, including user defaults, due to privacy reasons around device fingerprinting. This will impact Multiplatform Settings users if you're using NSUserDefaultsSettings.

See announcement here https://developer.apple.com/news/?id=z6fu1dcu
More details here https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278401

**Comment by vanniktech** [comment]
So if I write the exact same thing into a sqlite database, I don't need a privacy reason but for NSUserDefaults, we do? What am I overseeing?

**Reply by russhwolf** [reply]
I think the difference is, there are some globally readable settings in user defaults that can apparently be used for device fingerprinting, whereas if you're using sqlite you can only see the data you put there. But I definitely don't understand all the details of this yet.

**Reply by vanniktech** [reply]
There's some good information / sample here: https://github.com/onevcat/Kingfisher/issues/2122 - just stumbled upon this, but in general it seems like we're not the only ones who don't know what we're supposed to do
